src/collapse/collapse.js

```diff
@@ -11,28 +11,31 @@ angular.module('ui.bootstrap.collapse', [])
         }
 
         function expand() {
-          element.removeClass('collapse')
-            .addClass('collapsing')
-            .attr('aria-expanded', true)
-            .attr('aria-hidden', false);
+          $q.when(scope.$eval(attrs.expanding), function() {
```

Comment:
Good point about caching the expression.

FWIW, `$q.resolve(x, callback)` is more performant than `$q.resolve(x).then(callback)` since the latter waits for another tick.  But I can make the change if you'd prefer.